shared/clock/CMakeLists.txt

```diff
@@ -1 +1,8 @@
-add_library(nubots_clock OBJECT clock.cpp clock.hpp)
+# Create the clock library
+add_library(nubots_clock STATIC clock.cpp clock.hpp)
+target_compile_definitions(nubots_clock PUBLIC NUCLEAR_CUSTOM_CLOCK)
+set_property(TARGET nubots_clock PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
```suggestion
set_target_properties(nubots_clock PROPERTIES POSITION_INDEPENDENT_CODE ON)
```

Reply:
I found other occurrences of setting properties for targets with `set_property` and changed them too 👍